### PR TITLE
feat: Add CurseForge as a mod source

### PR DESCRIPTION
**Pull request by dyoung522**
## Summary

Adds CurseForge as a second mod source alongside NexusMods.

## Changes

- **New source implementation** for CurseForge API
- **CLI integration** with game resolution pattern (`--game` flag + `default_game` fallback)
- **Generic auth error messages** - no longer hardcoded to NexusMods
- **Documentation** for CurseForge setup and usage
- **Changelog** updated

## Testing

Full test suite passes ✅

### CODE REVIEW

File: internal/source/curseforge/client.go [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+		mod, err := c.GetMod(ctx, id)
+		if err != nil {
+			errs = append(errs, fmt.Errorf("mod %d: %w", id, err))
+			continue
+		}
+		mods = append(mods, *mod)
+	}
+
+	if len(errs) > 0 {
+		return mods, errors.Join(errs...)
```
[the remark this answers is not in the document]
Comment (dyoung522) [reply internal/source/curseforge/client.go]: Note: we have no need to support earlier versions.

**Comment by dyoung522** [comment]
Addressed the review comments in commit b41eaa1:

**Fixed:**
- **#3** docs/configuration.md: Added `deploy_mode` documentation with explanation and example
- **#4** importer.go: Moved regex to package-level `versionRegexImporter` variable
- **#7** curseforge.go: Using `errors.Join` consistently, removed custom `joinErrors` function
- **#8** client.go: Added `io.LimitReader` (10KB cap) for error response bodies
- **#10** auth.go: CurseForge API key validation now makes a test call to `GetGames()` instead of just checking length

**Responses to other comments:**

**#5 (import.go - CurseForge hardcoded):** The CurseForge-first matching is intentional for this feature. CurseForge has excellent mod fingerprinting via manifest hashes, making it the most accurate source for auto-detection. If CurseForge is not configured for a game, the lookup fails gracefully and falls through to filename-based matching. A future enhancement could iterate all sources, but for v1 this provides the best UX for the primary use case (Minecraft modding).

**#6 (curseforge.go - version comparison):** Agreed this is simplistic, but acceptable for v1. CurseForge "versions" are often file names rather than semantic versions, making proper semver comparison unreliable. The current logic catches the common case (new file available). Can revisit with a proper version parsing library in a future PR.

**#9 (importer.go - ScanModPath tests):** Will add test coverage in a follow-up PR. The scanning logic has some platform-specific edge cases (symlinks, permissions) that warrant dedicated test fixtures.
